qiita_db/software.py

```diff
@@ -537,29 +597,187 @@ def command(self):
         qiita_db.software.Command
             The command that this parameter set belongs to
         """
-        return self._command
+        with qdb.sql_connection.TRN:
+            sql = """SELECT command_id
+                     FROM qiita.default_parameter_set
+                     WHERE default_parameter_set_id = %s"""
+            qdb.sql_connection.TRN.add(sql, [self.id])
+            return Command(qdb.sql_connection.TRN.execute_fetchlast())
 
-    def to_str(self):
-        """Generates a string with the parameter values
+
+class Parameters(object):
+    """Represents an instance of parameters used to process an artifact
+
+    This class should never be isntantiated using the __init__ method. The
```

Comment:
I don't know how to do it. The main issue is that the `classmethods` are actually instantiating the object at the end, so those will raise the warning, although the user will be using the correct methods. Do you have any idea on how to do it?